QgisModelBaker/gui/panel/dataset_combobox.py

```diff
@@ -0,0 +1,166 @@
+        if self.isEnabled():
+            self.currentIndexChanged.disconnect(self.store_basket_tid)
+            self.setEnabled(False)
+
+        if not layer or not layer.dataProvider().isValid():
+            return
+
+        source_name = layer.dataProvider().name()
+        source = QgsDataSourceUri(layer.dataProvider().dataSourceUri())
+        schema_identificator = self.make_schema_identificator(source_name, source)
+        layer_model_topic_name = QgsExpressionContextUtils.layerScope(layer).variable('interlis_topic')
+
+        # set the filter of the model according the current uri_identificator
+        schema_topic_identificator = f"{schema_identificator}_{slugify(layer_model_topic_name)}"
+        self.filtered_model.setFilterFixedString(schema_topic_identificator)
+
+        if not self.basket_model.schema_baskets_loaded(schema_identificator):
+            mode = ''
+            configuration = Ili2DbCommandConfiguration()
+            if source_name == 'postgres':
+                mode = DbIliMode.pg
+                configuration.dbhost = source.host()
+                configuration.dbusr = source.username()
+                configuration.dbpwd = source.password()
+                configuration.database = source.database()
+                configuration.dbschema = source.schema()
+            elif source_name == 'ogr':
+                mode = DbIliMode.gpkg
+                configuration.dbfile = source.uri().split('|')[0].strip()
+            elif source_name == 'mssql':
+                mode = DbIliMode.mssql
+                configuration.dbhost = source.host()
+                configuration.dbusr = source.username()
+                configuration.dbpwd = source.password()
+                configuration.database = source.database()
+                configuration.dbschema = source.schema()
+            
+            if mode:   
+                db_factory = self.db_simple_factory.create_factory( mode ) 
+                config_manager = db_factory.get_db_command_config_manager(configuration)
+                self.basket_model.reload_schema_baskets(db_factory.get_db_connector( config_manager.get_uri(), configuration.dbschema), schema_identificator)
+
+        if self.filtered_model.rowCount():
+            self.currentIndexChanged.connect(self.store_basket_tid)
+            self.set_index(schema_topic_identificator)
+            self.setEnabled(True)
+
+    def set_index(self, schema_topic_identificator):
+        current_basket_tid = QgsExpressionContextUtils.projectScope(QgsProject.instance()).variable(schema_topic_identificator)
+        matches = self.filtered_model.match(self.filtered_model.index(0, 0), int(DatasetSourceModel.Roles.BASKET_TID), current_basket_tid, 1, Qt.MatchExactly)
+        if matches:
+            self.setCurrentIndex(matches[0].row())
+
+    def make_schema_identificator(self, source_name, source):
+        if source_name == 'postgres' or source_name == 'mssql':
```

Comment:
Could we move this code to the connector?

Reply:
Hm... we don't create this identification with information we got from the db (and it's connector functions) but from the layers and the data-sources instead. At this point we do not define any database configurations and connectors.

Reply:
I'd like to have all database specific code in one place. We can also put it into a utils file somewhere. If we ever add an oracle (or whatever) backend, we know where to change things.

Reply:
Good point. And our old teacher Mr. Vögelin would be happy about Oräcle ;-)

Reply:
utils/db_utils?
I prefer to have it there instead of db_connector.

Reply:
then we can put the only function in mssql_utils.py there as well..

Reply:
Maybe not merging it with the mssql_utils since otherwise we require always `pyodb`.